src/nm-ssh-service.c

```diff
@@ -515,7 +515,7 @@ send_network_config (NMSshPlugin *plugin)
 
 	/* replace default route? */
 	if (io_data->no_default_route) {
-		g_hash_table_insert (config, NM_VPN_PLUGIN_IP4_CONFIG_NEVER_DEFAULT, bool_to_gvalue (TRUE));
+		g_hash_table_insert (ip4config, NM_VPN_PLUGIN_IP4_CONFIG_NEVER_DEFAULT, bool_to_gvalue (TRUE));
```

Comment:
So look at the `#ifdef`s, they are a bit confusing, but if you compile with no IPv6 enabled, then `config = ip4config`. And in this case your change has no effect.

Last I saw, the only key to disable/enable default route in NM was called `NM_VPN_PLUGIN_IP4_CONFIG_NEVER_DEFAULT`. So in the case where you have IPv6 enabled, but don't want to have a default route, we disable it in the global config. I'll have to see if there's a new key called `NM_VPN_PLUGIN_IP6_CONFIG_NEVER_DEFAULT`, then this behaviour can be more precise.